controller/src/controller/main_systems/server.py

```diff
@@ -386,6 +396,27 @@ async def _set_stim_status(self, comm: dict[str, Any]) -> None:
 
         await self._to_monitor_queue.put(comm)
 
+    @mark_handler
+    async def _set_offline_state(self, comm: dict[str, Any]) -> None:
+        """Initiate or terminate system offline mode."""
+
+        system_state = self._get_system_state_ro()
+        system_status = system_state["system_status"]
+        incoming_offline_state = comm["offline_state"]
+
+        comms_to_ignore = (
```

Comment:
A little bit of a nitpick, but it looks to me like what this is doing is ignoring the command if the incoming state is the same as the current state, so the logic could just be written that way:
```
if incoming_offline_state is self._get_offline_state():
   return  # nothing to do here
```